tests/kubernetes-plugins/full.bats

```diff
@@ -0,0 +1,168 @@
+    helm repo update --fail-on-repo-update-fail
+
+    FLUENTBIT_ENV_VARS="env[0].name=TEST_NAMESPACE,env[0].value=${TEST_NAMESPACE},env[1].name=NODE_IP,env[1].valueFrom.fieldRef.fieldPath=status.hostIP"
+    helm upgrade --install --debug --create-namespace --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit \
+        --values ${BATS_TEST_DIRNAME}/resources/fluentbit-full.yaml \
+        --set image.repository=${FLUENTBIT_IMAGE_REPOSITORY},image.tag=${FLUENTBIT_IMAGE_TAG},${FLUENTBIT_ENV_VARS} \
+        --values "$HELM_VALUES_EXTRA_FILE" \
+        --timeout "${HELM_FB_TIMEOUT:-5m0s}" \
+        --wait
+}
+
+teardown_file() {
+    if [[ "${SKIP_TEARDOWN:-no}" != "yes" ]]; then
+        helm uninstall fluent-bit -n $TEST_NAMESPACE
+        run kubectl delete namespace "$TEST_NAMESPACE"
+        rm -f ${HELM_VALUES_EXTRA_FILE}
+    fi
+}
+
+setup() {
+    FLUENTBIT_POD_NAME=""
+    TEST_POD_NAME=""
+}
+
+teardown() {
+    if [[ "${SKIP_TEARDOWN:-no}" != "yes" ]]; then
+        if [[ ! -z "$TEST_POD_NAME" ]]; then
+            run kubectl delete pod $TEST_POD_NAME --grace-period 0
+        fi
+    fi
+}
+
+function set_fluent_bit_pod_name() {
+    try "at most 30 times every 2s " \
+        "to find 1 pods named 'fluentbit-ci-tests' " \
+        "with 'status' being 'Running'"
+    
+    FLUENTBIT_POD_NAME=$(kubectl get pods -n "$TEST_NAMESPACE" -l "app.kubernetes.io/name=fluent-bit" --no-headers | awk '{ print $1 }')
+    if [ -z "$FLUENTBIT_POD_NAME" ]; then
+        fail "Unable to get running fluent-bit pod's name"
+    fi
+}
+
+
+function create_test_pod() {
+    TEST_POD_NAME="$1"
+    # The hello-world-1 container MUST be on the same node as the fluentbit worker, so we use a nodeSelector to specify the same node name
+    run kubectl get pods $FLUENTBIT_POD_NAME -n $TEST_NAMESPACE -o jsonpath='{.spec.nodeName}'
+    assert_success
+    refute_output ""
+    node_name=$output
+
+    kubectl run -n $TEST_NAMESPACE $TEST_POD_NAME --image=docker.io/library/alpine:latest -l "this_is_a_test_label=true" \
+        --overrides="{\"apiVersion\":\"v1\",\"spec\":{\"nodeSelector\":{\"kubernetes.io/hostname\":\"$node_name\"}}}" \
+        --command -- sh -c 'while true; do echo "hello world"; sleep 1; done'
+
+    try "at most 30 times every 2s " \
+        "to find 1 pods named '$TEST_POD_NAME' " \
+        "with 'status' being 'Running'"
+    
+    # We are sleeping here specifically for the Fluent-Bit's tail input's
+    # configured Refresh_Interval to have enough time to detect the new pod's log file
+    # and to have processed part of it.
+    # A future improvement instead of sleep could use fluentbit's metrics endpoints
+    # to know the tail plugin has processed records
+    sleep 10
+}
+
+function check_fluent_output_has_pod_labels() {
+    HAS_POD_LABELS=$1
+    run kubectl logs -l "app.kubernetes.io/name=fluent-bit" -n "$TEST_NAMESPACE" --tail=1
+    assert_success
+    refute_output ""
```

Comment:
Are you just checking for non-empty here?

Reply:
yes